### DISCUSSION TITLE
How to add App Icon to window title bar?

### DISCUSSION THREAD

**Discussion by santokalayil** [fltk-rs/fltk-rs#506] [Q&A] [answered]
FLTK-Rust is a great GUI creation library... But It would be incomplete in the absense of setting App Icon to the window.. 
How to add App Icon to window title bar?

**Comment by MoAlyousef** [comment, marked as answer]
Hi
Thank you.
You can use the WindowExt::set_icon method:
```rust
let mut win = window::Window::default().with_size(600, 400).with_label("My Window");
win.end();
win.show();
let image = image::PngImage::load("assets/some_icon.png")?;
win.set_icon(Some(image));
```

**Reply by M4SSD35TRUCT10N** [reply]
Splendid, this would've been my next question.

**Reply by santokalayil** [reply]
>SHOWS ERROR HERE

use fltk::{app::*, button::*, frame::*, window::*, image::*};

fn main() {
    let app = App::default();
    let mut win = Window::new(100, 100, 400, 300, "Santo's App",);
    // let mut win = window::Window::default().with_size(600, 400).with_label("Ecclesiastics");

    // let image = image::PngImage::load("assets/some_icon.png")?;

    let image = PngImage::load("app_icon.png");
    win.set_icon(Some(image));
    // win.set_icon(Some(image));
    let mut frame = Frame::new(0, 0, 400, 200, "");
    let mut but = Button::new(160, 210, 80, 40, "Click me!");
    win.end();
    win.show();
    but.set_callback(move || frame.set_label("You clicked the Button!"));
    app.run().unwrap();

> ERROR
error[E0277]: the trait bound `std::result::Result<fltk::image::PngImage, fltk::FltkError>: fltk::ImageExt` is not satisfied
  --> src\main.rs:11:18
   |
11 |     win.set_icon(Some(image));
   |                  ^^^^^^^^^^^ the trait `fltk::ImageExt` is not implemented for `std::result::Result<fltk::image::PngImage, fltk::FltkError>`

error: aborting due to previous error

**Reply by santokalayil** [reply]
**Actually I figured it out right now...**

`use fltk::{app::*, button::*, frame::*, window::*, image::*};  //

fn main() {
    let app = App::default();
    let mut win = Window::new(100, 100, 400, 300, "Santo's App",);
    let image = PngImage::load(&std::path::Path::new("app_icon.png")).unwrap();
    win.set_icon(Some(image));
    let mut frame = Frame::new(0, 0, 400, 200, "");
    let mut but = Button::new(160, 210, 80, 40, "Click me!");
    win.end();
    win.show();
    but.set_callback(move || frame.set_label("You clicked the Button!"));
    app.run().unwrap();
}`

**Reply by M4SSD35TRUCT10N** [reply]
Could you mark the answer from @MoAlyousef as correct? That would be super nice and convenient.
